### PR TITLE
v1.21.0 dashboard performance fix

### PR DESCRIPTION
**Pull request by jacekkolasa**
##### Summary

Main issue: fix race condition in intersection observer. UseEffect was firing later than intersection observer making sequential callback, which caused unsync

Secondary:
remove profile flag from build task (that looks scary, but it had minor impact on performance when i tested it)

cancel request when chart goes out of view

### CODE REVIEW

File: web/gui/README.md [hunk 1]
```diff
@@ -31,6 +31,12 @@ settings](../server/#other-netdataconf-web-section-options) in the
 <iframe width="720" height="405" src="https://www.youtube.com/embed/Ob6-Wkb6ZBA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 </details>
 
+## New React Dashboard
+
+We've been working on refactoring the standard dashboard code to simplify the process of adding new features and fixing bugs. 
+
+The new dashboard is added to the codebase, but the old version is still here. If you want to use it, run `/usr/libexec/netdata/netdata-switch-dashboard.sh  classic`. To switch back, do `netdata-switch-dashboard.sh  react`. Currently, the refactored dashboard doesn't include any noticeable changes in UI.
```
Comment (joelhans) [inline web/gui/README.md:38]:
```suggestion
The new dashboard is added to the codebase, but the old version is still here. If you want to switch back to the old version, run `/usr/libexec/netdata/netdata-switch-dashboard.sh  classic`. To switch back to the React version, run `/usr/libexec/netdata/netdata-switch-dashboard.sh  react`. Currently, the React dashboard doesn't include any noticeable changes in user interface (UI) or features.
```

File: web/gui/README.md [hunk 2]
```diff
@@ -31,6 +31,12 @@ settings](../server/#other-netdataconf-web-section-options) in the
 <iframe width="720" height="405" src="https://www.youtube.com/embed/Ob6-Wkb6ZBA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 </details>
 
+## New React Dashboard
```
Comment (joelhans) [inline web/gui/README.md:34]:
```suggestion
## New React dashboard
```

**Comment by prologic** [comment]
@jacekkolasa This can be closed without merge in favor of #8664 right?

**Comment by jacekkolasa** [comment]
closing, this should be chosen instead - https://github.com/netdata/netdata/pull/8664

**Pull request closed without merging**